Lib/tkinter/test/test_tkinter/test_misc.py

```diff
@@ -1,10 +1,26 @@
 import unittest
+import subprocess
+import sys
+import threading
 import tkinter
 from test import support
 from tkinter.test.support import AbstractTkTest
 
 support.requires('gui')
 
+def test_thread_gc():
+    def wait(obj):
+        ev1.wait()
+        ev2.set()
+    root = tkinter.Tk()
+    root.destroy()
+    ev1, ev2 = threading.Event(), threading.Event()
+    threading.Thread(target=wait, args=(root,)).start()
+    del root
+    ev1.set()
+    ev2.wait()
```

Comment:
I think `ev2` is unnecessary?  You could substitute it by just joining the thread, but I think that is unnecessary too, it's OK for the spawned thread to be the last one standing.